pipelines/tekton/gitops-update-pipeline/gitops-update-git-workspace-bike-rentals-pvc.yaml

```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 kind: PersistentVolumeClaim
 metadata:
-  name: git-workspace-pvc
+  name: git-workspace-bike-rentals
```

Comment:
Having separate workspaces might be better, but does it matter since the repo's in them get replaced every time a clone happens anyway? The repo's are also cloned to the workspace using their repo's name so there's no chance of collision too.

Reply:
There are a couple of reasons I think that separate workspaces would be better:

* The PVCs have the ReadWriteOnce access mode, which means that only one Pod (or PipelineRun) can use it at a time. This could cause PipelineRuns for the _other_ repo to block unnecessarily while waiting for the current run to complete (if there were many many different variations to the pipeline (i.e. repos), that got run a lot, that might slow things down. I realise it's an unlikely case, but blocking per repo rather than per workspace seems cleaner.
* Even though, as you say, the repos are cloned to the workspace using the repo name, that could still cause a collision in our examples, since both have the same repo name (just in different orgs), right? Granted, I think the access mode on the PVC would prevent that collision, given that only one PipelineRun should be able to mount the PVC anyway

I actually think using Tekton's ability to generate a PVC per PipelineRun might be even better, but maybe a bit wasteful if there's nothing to clean up old pipeline runs older than a certain age or outside the most recent N runs.

WDYT?

Reply:
You're totally right on both points, didn't think of that 👍. 
<br>

> I actually think using Tekton's ability to generate a PVC per PipelineRun might be even better

You mean [like this](
https://tekton.dev/docs/pipelines/workspaces/#volumeclaimtemplate)? I also like that approach.
<br>

> maybe a bit wasteful if there's nothing to clean up old pipeline runs

[OpenShift Pipelines to the rescue](https://docs.openshift.com/container-platform/4.13/cicd/pipelines/customizing-configurations-in-the-tektonconfig-cr.html#annotations-for-automatic-pruning-taskruns-pipelineruns_customizing-configurations-in-the-tektonconfig-cr). There's also a [cron job](https://gist.github.com/ctron/4764c0c4c4ea0b22353f2a23941928ad).

Reply:
Updated to use Tekton's PVC generation per run.